modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/cache/ResetLostPartitions.java

```diff
@@ -67,9 +85,47 @@ public class ResetLostPartitions extends AbstractCommand<Set<String>> {
         }
     }
 
+    /** Get a list of caches with LOST partitions in case if ALL MODE is specified. */
+    private void getAffectedCaches(GridClientConfiguration clientCfg, Logger logger) throws Exception {
+        logger.info("ALL mode is activated.");
+        logger.info("Looking for the caches with LOST partitions first...");
+
+        caches = new HashSet<>();
+        CacheDistributionTaskArg taskArg = new CacheDistributionTaskArg(null, null);
+        try (GridClient client = Command.startClient(clientCfg)) {
+            CacheDistributionTaskResult res = executeTaskByNameOnNode(client, CacheDistributionTask.class.getName(), taskArg, BROADCAST_UUID, clientCfg);
+            for (CacheDistributionNode node : res.jobResults()) {
+                for (CacheDistributionGroup group : node.getGroups()) {
+                    for (CacheDistributionPartition partition : group.getPartitions()) {
+                        if (partition.getState() == GridDhtPartitionState.LOST) {
+                            caches.add(group.getGroupName());
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        if (caches.size() > 0)
+            logger.info("The following caches have LOST partitions: " + CommandLogger.join(",", caches) + ".");
+        else
+            logger.info("No caches with LOST partition has been found found.");
+    }
+
     /** {@inheritDoc} */
     @Override public void parseArguments(CommandArgIterator argIter) {
-        caches = argIter.nextStringSet("Cache names");
+
+        String nextArg = argIter.nextArg("");
```

Comment:
It would be nice to provide a detailed message. Your implementation assumes that it should be a cache name, or `--all` option.

Reply:
Fixed